From 1c7bd7a7ff516c77ac97274027c6c61797bb625a Mon Sep 17 00:00:00 2001
From: Steven Barrett <steven@liquorix.net>
Date: Wed, 27 Jan 2021 14:27:35 -0600
Subject: [PATCH 5/8] ZEN: Fix mq-deadline scheduler renaming

In recent ZEN commit ("ZEN: Add CONFIG to rename the mq-deadline
scheduler"), we only handled the elevator name if mq-deadline, but not
its elevator or module alias.  Fix these two so that we can pick
mq-deadline-nodefault without errors.
---
 block/mq-deadline.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/block/mq-deadline.c b/block/mq-deadline.c
index 25fec9494..b2b2f7a15 100644
--- a/block/mq-deadline.c
+++ b/block/mq-deadline.c
@@ -792,6 +792,7 @@ static struct elevator_type mq_deadline = {
 	.elevator_attrs = deadline_attrs,
 #ifdef CONFIG_MQ_IOSCHED_DEADLINE_NODEFAULT
 	.elevator_name = "mq-deadline-nodefault",
+	.elevator_alias = "deadline-nodefault",
 #else
 	.elevator_name = "mq-deadline",
 	.elevator_alias = "deadline",
@@ -799,7 +800,11 @@ static struct elevator_type mq_deadline = {
 	.elevator_features = ELEVATOR_F_ZBD_SEQ_WRITE,
 	.elevator_owner = THIS_MODULE,
 };
+#ifdef CONFIG_MQ_IOSCHED_DEADLINE_NODEFAULT
+MODULE_ALIAS("mq-deadline-nodefault-iosched");
+#else
 MODULE_ALIAS("mq-deadline-iosched");
+#endif
 
 static int __init deadline_init(void)
 {
-- 
2.31.1.621.g97eea85a0a

